### PR TITLE
Michael/menu action buttons

### PR DESCRIPTION
**Pull request by msenejoa**

### CODE REVIEW

File: packages/databyss-ui/components/Editor/EditorMenu/EditorMenu.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        </SidebarButton>
+      </View>
+      <View
+        ml="smallNegative"
+        justifyContent="center"
+        height={space.menuHeight}
+      >
+        {actions && (
+          <Grid mb="none" rowGap="small" columnGap="small">
+            <View>
```
Comment (msenejoa) [inline packages/databyss-ui/components/Editor/EditorMenu/EditorMenu.js]: this needs to be wrapped in a <View/>

File: packages/databyss-ui/primitives/Button/SidebarButton.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      <Text variant="uiTextExtraSmall" color={buttonVariants[variant].color}>
+        {children}
+      </Text>
+    )
+  }
+  return (
+    <BaseControl
+      onPress={onPress}
+      rippleColor={buttonVariants[variant].rippleColor}
+      borderRadius="50%"
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Button/SidebarButton.js]: This has to be an Int or Float for native. So use `pxUnits()` and set it to half the height.
```
borderRadius={pxUnits(space.editorBlockMenuHeight / 2)}
width={pxUnits(space.editorBlockMenuHeight)}
height={pxUnits(space.editorBlockMenuHeight)}
```

File: packages/databyss-ui/primitives/Button/SidebarButton.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      height="20px"
+      width="20px"
+      justifyContent="center"
+      borderWidth="1px"
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Button/SidebarButton.js]: pxUnits

File: packages/databyss-ui/primitives/Text/RawHtml.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { variant, color, compose } from 'styled-system'
+import styled from '../styled'
+import IS_NATIVE from '../../lib/isNative'
+
+const variants = variant({
+  prop: 'variant',
+  scale: 'textVariants',
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Text/RawHtml.js]: There's a hack to get variants to parse styled-system theme values:
```
const variants = variant({
  prop: 'sizeVariant',
  scale: 'textVariants',
  variants: {
    // need one member to enable theming
    default: {},
  },
})
```

File: packages/databyss-ui/primitives/Text/Text.js [hunk 5]
```diff
@@ -7,7 +7,7 @@ const variants = variant({
   scale: 'textVariants',
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Text/Text.js]: add `variants:` key as above

File: packages/databyss-ui/primitives/Text/Text.js [hunk 6]
```diff
@@ -19,8 +19,14 @@ const Text = styled(
   )
 )
 
-export default ({ children, ...others }) => (
-  <Text variant="bodyNormal" {...others}>
+const Text = ({ children, color, inline, _html, ...others }) => (
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Text/Text.js]: Looks like we no longer need `inline` and `html` no with `RawHtml`

File: packages/databyss-ui/theming/space.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ const space = [0, 8, 16, 24, 32, 64]
 space.none = space[0]
 space.small = space[1]
 space.medium = space[3]
+space.menuHeight = space[4]
```
Comment (pzhine) [inline packages/databyss-ui/theming/space.js]: Let's rename to `space.editorBlockMenuHeight`

File: packages/databyss-ui/components/Editor/EditorBlock.js [hunk 8]
```diff
@@ -1,38 +1,61 @@
...
 
-const EditorBlock = ({ type, children }) => (
-  <Grid mb="medium" flexWrap="nowrap" columnGap="small" alignItems="baseline">
+const EditorBlock = ({ children, node }) => (
+  <Grid mb="medium" flexWrap="nowrap" columnGap="small">
     <View
       contentEditable="false"
       suppressContentEditableWarning
       css={{ userSelect: 'none' }}
+      width={1 / 10}
```
[the remark this answers is not in the document]
Comment (pzhine) [reply packages/databyss-ui/components/Editor/EditorBlock.js]: in your SidebarButton component, add:
```
export const buttonDimension = 40
```
and then use that to calculate widths, heights, etc

File: packages/databyss-ui/primitives/Text/RawHtml.js (hunk 4 above)
Comment (pzhine) [reply packages/databyss-ui/primitives/Text/RawHtml.js]: Let's just import the variants from Text

File: packages/databyss-ui/primitives/Button/SidebarButton.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+      borderColor={buttonVariants[variant].borderColor}
+      {...others}
+    >
+      <StyledView variant={variant}>{_children}</StyledView>
+    </BaseControl>
+  )
+}
+
+SidebarButton.defaultProps = {
+  variant: 'sidebarAction',
```
Comment (pzhine) [inline packages/databyss-ui/primitives/Button/SidebarButton.js]: let's call this variant "editorBlockMenuButton"

**Pull request closed without merging**